tests/Backend/Teradata/ImportTypedTableTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Keboola\Test\Backend\Teradata;
+
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Workspaces;
+use Keboola\TableBackendUtils\Column\Teradata\TeradataColumn;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+use Keboola\Test\Backend\Workspaces\ParallelWorkspacesTestCase;
+
+class ImportTypedTableTest extends ParallelWorkspacesTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $token = $this->_client->verifyToken();
+
+        if (!in_array('tables-definition', $token['owner']['features'])) {
+            $this->markTestSkipped(sprintf('Tables definition feature is not enabled for project "%s"', $token['owner']['id']));
+        }
+
+        $this->initEmptyTestBucketsForParallelTests();
+    }
+
+    public function testLoadTypedTablesConversionError(): void
+    {
+        $fullLoadFile = __DIR__ . '/../../_data/users.csv';
+
+        $bucketId = $this->getTestBucketId(self::STAGE_IN);
+        $payload = [
+            'name' => 'users-types',
+            'primaryKeysNames' => ['id'],
+            'columns' => [
+                ['name' => 'id', 'definition' => ['type' => 'INT', 'nullable' => false]],
+                ['name' => 'name', 'definition' => ['type' => 'VARCHAR']],
+                ['name' => 'city', 'definition' => ['type' => 'VARCHAR']],
+                ['name' => 'sex', 'definition' => ['type' => 'INT']],
+            ],
+        ];
+        $runId = $this->_client->generateRunId();
+        $this->_client->setRunId($runId);
+
+        // create table
+        $tableId = $this->_client->createTableDefinition($bucketId, $payload);
+
+        // block until async events are processed, processing in order is not guaranteed but it should work most of time
+        $this->createAndWaitForEvent((new \Keboola\StorageApi\Event())->setComponent('dummy')->setMessage('dummy'));
+
+        try {
+            // try import data with wrong types with full load
+            $this->_client->writeTableAsync(
+                $tableId,
+                new CsvFile($fullLoadFile),
+                [
+                    'incremental' => false,
+                ]
+            );
```

Comment:
nema tady byt fail? pokud jsi to kopiroval z jinych testu, tak to prosim dopln i tam

Reply:
jo, dobrej point. doplneno i do ostatnich